src/optimagic/optimization/internal_optimization_problem.py

```diff
@@ -704,6 +717,33 @@ def _pure_evaluate_fun_and_jac(
 
         return (algo_fun_value, out_jac), hist_entry, log_entry
 
+    def _check_infinite_gradients(
```

Comment:
I prefer

```python
def _assert_finite_jac(...)
 ...
```

instead of the negative form.

I would also speak of `jac` and not `gradient`, because we are dealing with the jacobian (it is only the gradient in the special case of scalar-valued function).